x-pack/plugins/security_solution/public/resolver/types.ts

```diff
@@ -238,31 +243,54 @@ export interface NodeEventsInCategoryState {
    * Flag for showing an error message when fetching additional related events.
    */
   error?: boolean;
+
+  /**
+   * Flag for showing a loading state in the related event panel.
+   */
+  loading?: boolean;
```

Comment:
This might overcomplicate things but could we infer that it is loading by `pendingRequest !== undefined`? That way we wouldn't need this variable too right?